app/services/users/cleanup_chat_channels.rb

```diff
@@ -0,0 +1,18 @@
+module Users
+  module CleanupChatChannels
+    def self.call(user)
+      user.chat_channels.each do |chat_channel|
+        chat_channel.remove_from_index!
+        chat_channel.chat_channel_memberships.each do |ccm|
+          ccm.remove_from_index!
+          ccm.destroy!
+        end
+
+        # We only destroy direct message channels, not open and invite-only ones
```

Comment:
@Zhao-Andy Your snippet in #3601 did not include a check list this, but I guess it's better to err on the side of caution here. Running the specs with the next line commented out would remove all open channels the banished user was in, which is probably not what we want.

Reply:
Ah yes, we should only `destroy` direct chat channels. I think we might also want to only `remove_from_index!` for direct chat channels as well. Most banished users won't be a part of any open channels, but better to be safe here.

Reply:
Ok, I'll move that below the `next unless chat_channel.direct?` guard clause too. 

Also, this snippet:

```ruby
        chat_channel.chat_channel_memberships.each do |ccm|
          ccm.remove_from_index!
          ccm.destroy!
        end
```

Won't this remove channel memberships for all other users if a banished account was part of an open channel? 

I'd rather play this safe, so I rewrote the whole method to do this in a two step process:

1. Get all direct channels and clean them up.
2. Iterate over the user's remaining memberships and clean them up, leaving the channels alone.